src/java/org/apache/cassandra/service/snapshot/TableSnapshot.java

```diff
@@ -490,8 +496,90 @@ private void loadMetadataFromManifest(File manifestFile)
 
         TableSnapshot build()
         {
+            maybeCreateOrEnrichManifest();
             return new TableSnapshot(keyspaceName, tableName, tableId, tag, createdAt, expiresAt, snapshotDirs, ephemeral);
         }
+
+        private void maybeCreateOrEnrichManifest()
+        {
+            // this is caused by not reading any manifest of a new format or that snapshot had none upon loading,
+            // that might be the case when upgrading e.g. from 4.0 where basic manifest is created when taking a snapshot,
+            // so we just go ahead and enrich it in each snapshot dir
+            if (createdAt != null)
```

Comment:
Nit: Is it possible to check what of the two cases it is? Maybe adding a logger debug line stating it may be useful?